### PR TITLE
Improve documentation for stats

### PR DESCRIPTION
**Pull request by konsumlamm**

### CODE REVIEW

File: lib/pure/stats.nim [hunk 1]
```diff
@@ -9,65 +9,70 @@
...
-##  rs.push(MySeqOfData)
-##  rs.mean()
-##  rs.variance()
-##  rs.skewness()
-##  rs.kurtosis()
+## efficient to push the data once to a `RunningStat` object and then
+## call the numerous statistical procs for the `RunningStat` object:
+
+runnableExamples:
+  template `===`(a, b: float): bool = abs(a - b) < 1e-9
```
Comment (timotheecour) [inline lib/pure/stats.nim]: how about:
```nim
  import std/math
  template `~=`(a, b: float): bool = approxEqual(a, b)
```

File: lib/pure/stats.nim [hunk 2]
```diff
@@ -9,65 +9,70 @@
...
-## ``RunningRegress`` calculates for two sets of data
-## - n
+## `RunningRegress` calculates for two sets of data
+## - n (data count)
 ## - slope
 ## - intercept
 ## - correlation
 ##
-## Procs have been provided to calculate statistics on arrays and sequences.
+## Procs are provided to calculate statistics on `openArray`s (arrays and sequences).
```
Comment (timotheecour) [inline lib/pure/stats.nim]:
```suggestion
## Procs are provided to calculate statistics on `openArray`s.
```

other types can be converted to `openArray` (eg string) but more importantly we shouldn't repeat definitions. Future work might turn those into clickable links for jump to definition anyways.

**Review by timotheecour — APPROVED** [review]
LGTM after nits

File: lib/pure/stats.nim [hunk 3]
```diff
@@ -9,65 +9,72 @@
 
 ## Statistical analysis framework for performing
 ## basic statistical analysis of data.
-## The data is analysed in a single pass, when a data value
-## is pushed to the ``RunningStat`` or ``RunningRegress`` objects
+## The data is analysed in a single pass, when it
+## is pushed to a `RunningStat` or `RunningRegress` objects.
```
Comment (Araq) [inline lib/pure/stats.nim]:
```suggestion
## is pushed to a `RunningStat` or `RunningRegress` object.
```

**Pull request merged**